yt_dlp/extractor/radiojavan.py

```diff
@@ -10,20 +10,276 @@
+            webpage, 'artist', fatal=False)
+
+        # For Podcasts, RJ displays song title with "artist" class!!
+        song = self._search_regex(
+            r'<span class="artist">([^<]+?)<',
+            webpage, 'song', fatal=False)
+
+        title = f'{artist} - {song}'
+        alt_title = self._og_search_title(webpage)
+        thumbnail = self._og_search_thumbnail(webpage)
+
+        upload_date = unified_strdate(self._search_regex(
+            r'class="dateAdded">Date added: ([^<]+)<',
+            webpage, 'upload date', fatal=False))
+
+        view_count = str_to_int(self._search_regex(
+            r'class="views">Plays: ([\d,]+)',
+            webpage, 'view count', fatal=False))
+        like_count = str_to_int(self._search_regex(
+            r'class="rating">([\d,]+) likes',
+            webpage, 'like count', fatal=False))
+
+        return {
+            'id': podcast_id,
+            'title': title,
+            'alt_title': alt_title,
+            'track': song,
+            'artist': artist,
+            'url': f'{download_host}/media/{mp3_url}.mp3',
+            'thumbnail': thumbnail,
+            'upload_date': upload_date,
+            'view_count': view_count,
+            'like_count': like_count,
+        }
+
+
+class RadioJavanMp3IE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?radiojavan\.com/mp3s/(mp3/(?P<id>[^/]+)|playlist_start\?id=(?P<playlist_id>[^/]+)&index=(?P<index>[\d]+))/?'
+    IE_NAME = 'radiojavan:mp3'
+    _TESTS = [{
+        'url': 'https://www.radiojavan.com/mp3s/mp3/Ell3-Baroon',
+        'md5': 'cb877362f8e8fabb1aad6e2f1bf1bf97',
+        'info_dict': {
+            'id': 'Ell3-Baroon',
+            'ext': 'mp3',
+            'title': 'Ell3 - Baroon',
+            'alt_title': 'Ell3 - Baroon Song | ال بارون',
+            'track': 'Baroon',
+            'artist': 'Ell3',
+            'thumbnail': r're:^https?://.*\.jpe?g$',
+            'upload_date': '20221226',
+            'view_count': int,
+            'like_count': int,
+        }
+    }, {
+        'url': 'https://www.radiojavan.com/mp3s/playlist_start?id=6449cdabd351&index=0',
+        'info_dict': {
+            'id': str,  # This is a playlist item and could simply change in future, thus we validate by type
+            'title': str,
+            'alt_title': str,
+            'track': str,
+            'artist': str,
+            'thumbnail': r're:^https?://.*\.jpe?g$',
+            'upload_date': str,
+            'view_count': int,
+            'like_count': int,
+        },
+        'params': {
+            # This is playlist item and it could change later,
+            # so we provide an output name so we will not get 'info file not found' as we validating title by type
+            'outtmpl': 'RJPlaylistMP3',
+            # It is a random track, thus, we are unable to validate it's md5.
+            'skip_download': True,
+            'ignore_no_formats_error': True,
+        }
+    }]
```

Comment:
This isn't actually testing anything? Does the first item change very often? If not, just hardcode it